src/main/java/org/sonarlint/intellij/ui/CurrentFilePanel.java

```diff
@@ -150,12 +151,18 @@ public void update(@Nullable VirtualFile file, @Nullable Collection<LiveIssue> i
       return;
     }
     if (issues == null) {
-      statusText.setText("No analysis done on the current opened file");
       var templateText = analyzeCurrentFileAction.getTemplateText();
-      if (templateText != null) {
-        statusText.appendLine(templateText, SimpleTextAttributes.LINK_PLAIN_ATTRIBUTES,
-          ignore -> ActionUtil.invokeAction(analyzeCurrentFileAction, this, CurrentFilePanel.SONARLINT_TOOLWINDOW_ID, null, null));
+
+      if (getService(project, AnalysisReadinessCache.class).isReady()) {
+        statusText.setText("No analysis done on the current opened file");
+        if (templateText != null) {
+          statusText.appendLine(templateText, SimpleTextAttributes.LINK_PLAIN_ATTRIBUTES,
+            ignore -> ActionUtil.invokeAction(analyzeCurrentFileAction, this, CurrentFilePanel.SONARLINT_TOOLWINDOW_ID, null, null));
+        }
+      } else {
+        statusText.setText("Waiting for SonarLint to be ready");
```

Comment:
Have you tested if you see this message? This method is normally called when receiving updates IIRC

Reply:
Yes, if backend is not ready this message appears